lib/index.js

```diff
@@ -86,7 +86,7 @@ Mongoose.prototype.STATES = STATES;
  * Currently supported options are:
  * - 'debug': prints the operations mongoose sends to MongoDB to the console
  * - 'bufferCommands': enable/disable mongoose's buffering mechanism for all connections and models
- * - 'useFindAndModify': true by default. Set to `false` to make `findOneAndUpdate()` and `findOneAndRemove()` use native `findOneAndUpdate()` rather than `findAndModify()`.
+ * - 'useFindAndModify': false by default. Set to `true` to make `findOneAndUpdate()` and `findOneAndRemove()` use legacy `findAndModify()` rather than native methods`.
```

Comment:
I'm wary of changing `useFindAndModify` to false by default in a minor release, especially in light of the issue with `overwrite: true` that you found. I was not aware of that difference, which makes me wonder if there are other compatibility concerns that we're missing.

Reply:
It makes sense so reverting this change. Developers should opt in to suppress deprecation warning and test if it breaks anything at this moment.